platform/security/certificate/security-ocsp-checker/src/test/java/org/codice/ddf/security/ocsp/checker/OcspCheckerTest.java

```diff
@@ -0,0 +1,291 @@
+    Certificate certificate = getBouncyCastleCertificate();
+
+    OCSPReq ocspReq = ocspChecker.generateOcspRequest(certificate);
+    assertNotNull(ocspReq);
+
+    Assert.assertEquals(
+        ocspReq.getRequestList()[0].getCertID().getSerialNumber(),
+        certificate.getSerialNumber().getValue());
+  }
+
+  @Test
+  public void testSendingOcspRequest() throws Exception {
+    ClientFactoryFactory clientFactoryFactory = mockClientFactoryResponse();
+    OcspChecker ocspChecker = new OcspChecker(clientFactoryFactory, eventAdmin);
+    ocspChecker.setOcspServerUrl("https://testurl:8993");
+
+    Certificate certificate = getBouncyCastleCertificate();
+    OCSPReq ocspReq = ocspChecker.generateOcspRequest(certificate);
+
+    Response response = ocspChecker.sendOcspRequest(ocspReq);
+
+    assertNotNull(response);
+  }
+
+  @Test
+  public void testValidatingOCSPResponseWithUnknownStatus() throws Exception {
+    OcspChecker ocspChecker = new OcspChecker(factory, eventAdmin);
+
+    OCSPResp resp = mockOcspResponse(new UnknownStatus());
+    boolean response = ocspChecker.isOCSPResponseValid(resp);
+    assertFalse(response);
+  }
+
+  @Test
+  public void testValidatingOCSPResponseWithRevokedStatus() throws Exception {
+    OcspChecker ocspChecker = new OcspChecker(factory, eventAdmin);
+
+    OCSPResp resp = mockOcspResponse(new RevokedStatus(new Date(), 0));
+    boolean response = ocspChecker.isOCSPResponseValid(resp);
+    assertFalse(response);
+  }
+
+  @Test
+  public void testValidatingOCSPResponseWithNullBasicResponse() throws Exception {
+    OcspChecker ocspChecker = new OcspChecker(factory, eventAdmin);
+
+    OCSPResp resp = mock(OCSPResp.class);
+    when(resp.getResponseObject()).thenReturn(null);
+
+    boolean response = ocspChecker.isOCSPResponseValid(resp);
+    assertFalse(response);
+  }
+
+  @Test
+  public void testValidatingOCSPResponseWithNullResponse() throws Exception {
+    OcspChecker ocspChecker = new OcspChecker(factory, eventAdmin);
+
+    OCSPResp resp = mock(OCSPResp.class);
+    BasicOCSPResp basicOCSPResp = mock(BasicOCSPResp.class);
+    when(basicOCSPResp.getResponses()).thenReturn(null);
+    when(resp.getResponseObject()).thenReturn(basicOCSPResp);
+
+    boolean response = ocspChecker.isOCSPResponseValid(resp);
+    assertFalse(response);
+  }
+
+  @Test
+  public void testValidatingOCSPResponseWithNullStatus() throws Exception {
+    OcspChecker ocspChecker = new OcspChecker(factory, eventAdmin);
+
+    OCSPResp resp = mockOcspResponse(null);
+    boolean response = ocspChecker.isOCSPResponseValid(resp);
+    assertTrue(response);
+  }
+
+  @Test(expected = IOException.class)
+  public void testPostEvent() throws Exception {
+    OcspChecker ocspChecker = new OcspChecker(factory, eventAdmin);
+    Response response = mock(Response.class);
+
+    ocspChecker.createOcspResponse(response);
+    verify(eventAdmin, times(1)).postEvent(any());
```

Comment:
✏️ If `createOcspResponse` on line 168 is throwing an exception as the test expects, then this line will never get called and can be removed.